harness/determined/common/experimental/model.py

```diff
@@ -301,29 +305,32 @@ def set_labels(self, labels: List[str]) -> None:
         if not isinstance(labels, Iterable):
             raise ValueError(f"set_labels() requires a list of strings as input but got: {labels}")
 
-        self.labels = list(labels)
+        labels = list(labels)
 
-        req = bindings.v1PatchModel(labels=self.labels)
+        req = bindings.v1PatchModel(labels=labels)
         bindings.patch_PatchModel(self._session, body=req, modelName=self.name)
 
+        self.labels = labels
+
     def set_description(self, description: str) -> None:
-        self.description = description
-        req = bindings.v1PatchModel(description=self.description)
+        req = bindings.v1PatchModel(description=description)
         bindings.patch_PatchModel(self._session, body=req, modelName=self.name)
 
+        self.description = description
+
     def archive(self) -> None:
         """
         Sets the model's state to archived
         """
-        self.archived = True
         bindings.post_ArchiveModel(self._session, modelName=self.name)
+        self.archived = True
```

Comment:
would this update archived state on a POST failure?

Reply:
I expect failures in the bindings call to raise an exception. Do you think that's incomplete, or is there something else I might be missing?